### ISSUE TITLE
Web Support

### ISSUE THREAD

**Issue by erabti** [marcglasberg/network_to_file_image#21]
Does this plugin support web and if so how?

**Comment by jodymac** [comment]
Would love to see this as well so I don't have to use a different package for web. I'm already in the directory structure where I store my images as I pull them down from my website to populate the mobile devices.  My use case would be that it doesn't need to store/cache the data in the web platform,  just display from the url, hopefully a relative url.

**Comment by marcglasberg** [comment]
This widget downloads images to the local disk. How would that work for the web? There's no even a local disk available. 

@jodymac Could you explain in more detail your proposal? I am not sure I follow.

**Comment by jodymac** [comment]
You're right and I didn't explain well.  To rephrase, I believe I should have said it like this.  I use your plugin for my apps, when I create a website from the existing code, the website breaks because of the plugin and I have to rewrite the logic.  Not to hard to change to using a web compatible method, but would like to use yours only.  Can you make the plugin detect if web and only use the download feature and ignore the saving to disk and checking cache.

In hindsight and having configured a few web apps to use different methods while still using your plugin for devices, the benefit I receive from that function is probably bloatware for the plugin and maybe not worth the time it would require you to have to implement that feature.  Was just trying to use one plugin on all platforms.

**Comment by marcglasberg** [comment]
@jodymac I was looking into this. I wanted to import different packages, using conditional imports, where one library in the package uses `dart:io`, and another which uses `dart:js`, and then import whichever of these is supported:

```dart
import "file_loader.dart" // Version which just throws UnsupportedError
  if (dart.library.io) "file_loader_io.dart"
  if (dart.library.js) "file_loader_js.dart";
```

The problem is that I need to give these libraries the same API - so that no matter which library is imported, the code that uses it is still valid. But the `NetworkToFileImage` API itself uses the `File` class, which is from `dart:io`. So I can't remove `dart:io`, unless I change the public API to receive a `String` instead.

Would you have a different suggestion?
